### DISCUSSION TITLE
How do I access AZ::DX12::ShaderStageFunction in AtomSampleViewer Components?

### DISCUSSION THREAD

**Discussion by ohmaya** [o3de/o3de#13326] [Help Wanted]
Or, even cannot access AZ::DX12::ShaderStageFunction in AZ::RHI::ShaderStageFunction?

Thanks

**Comment by moudgils** [comment]
Shaderstagefunction contains shader bytecode emitted by the shader pipeline which is run by Asset processor. In order to access it you just need to load the shader and should be able to access RHI::ShaderStageFunction* and if it is a dx12 shader you can cast it to AZ::DX12::ShaderStageFunction*

```
 //! Returns the shader stage function associated with the provided stage enum value.
 const RHI::ShaderStageFunction* GetShaderStageFunction(RHI::ShaderStage shaderStage) const;
```
